python/cuml/test/stemmer_tests/test_steps.py

```diff
@@ -0,0 +1,238 @@
+#
+# Copyright (c) 2019-2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import cudf
+from cuml.preprocessing.text.stem.porter_stemmer import PorterStemmer
+
+
+def test_step1a():
+    word_str_ser = cudf.Series(
+        ["caresses", "ponies", "ties", "caress", "cats"]
+    )
+
+    st = PorterStemmer()
+    got = st._step1a(word_str_ser)
+
+    expect = ["caress", "poni", "tie", "caress", "cat"]
+    assert list(got.to_pandas().values) == expect
+
+
+def test_step1b():
+    word_str_ser_ls = [
+        "feed",
+        "agreed",
+        "plastered",
+        "bled",
+        "motoring",
+        "sing",
+        "conflated",
+        "troubled",
+        "sized",
+        "hopping",
+        "tanned",
+        "falling",
+        "hissing",
+        "fizzed",
+        "failing",
+        "filing",
+    ]
+
+    expected = [
+        "feed",
+        "agree",
+        "plaster",
+        "bled",
+        "motor",
+        "sing",
+        "conflate",
+        "trouble",
+        "size",
+        "hop",
+        "tan",
+        "fall",
+        "hiss",
+        "fizz",
+        "fail",
+        "file",
+    ]
+
+    word_str_ser = cudf.Series(word_str_ser_ls)
+    st = PorterStemmer()
+    got = st._step1b(word_str_ser)
+
+    assert list(got.to_pandas().values) == expected
+
+
+def test_step1c():
+    word_str_ser_ls = ["happy", "sky", "enjoy", "boy", "toy", "y"]
+    word_str_ser = cudf.Series(word_str_ser_ls)
```

Comment:
Should we generate a `can_replace_mask` for the `test_step*` tests as well?

Reply:
Resolved by : https://github.com/rapidsai/cuml/pull/2476/commits/9f886fa996e422df6a10d03ffbb14b075512ff8b